src/ir/module-splitting.cpp

```diff
@@ -369,17 +428,57 @@ void ModuleSplitter::setupTablePatching() {
     }
   });
 
+  if (replacedElems.size() == 0) {
+    // No placeholders to patch out of the table
+    return;
+  }
+
+  if (tableManager.activeBase.global) {
+    assert(primary.table.segments.size() == 1 &&
+           "Unexpected number of segments with non-const base");
+    assert(secondary.table.segments.size() == 0);
+    // Since addition is not currently allowed in initializer expressions, we
+    // need to start the new secondary segment where the primary segment starts.
+    // The secondary segment will contain the same primary functions as the
+    // primary module except in positions where it needs to overwrite a
+    // placeholder function. All primary functions in the table therefore need
```

Comment:
Another option might be to put the secondary segment before the primary one. Then it doesn't matter what it has in places defined in the primary one, which will overwrite those.

Reply:
The secondary segment is in the secondary module, so there's no way to load it before the primary segment.

Reply:
Oh, right...